### PR TITLE
Adds broker job stream service

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR adds a server endpoint for the job stream API, as well as the minimum communication protocol. The endpoint can receive requests to:

1. Add streams to the registry
2. Remove a specific stream from the registry
3. Remove all streams associated with a member from the registry

Additionally, when a member is removed from the cluster, all streams associated with it are removed. In the future we can think about handling `REACHABILITY_CHANGED` to temporarily disable certain streams without completely removing them.

The `LongPollingJobNotification` was replaced with scaffolding for the `JobGatewayStreamer`, but the functionality is largely the same. This will be expanded in a follow up PR.

## Related issues

related to #11707 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by npepinpe** [comment]
Seems like changing the schema version is a breaking change. I can see why you could think that. Do you think we shouldn't bump the version when we add new things, and only do it with breaking changes?

Anyway, other notes: I'm not sure I like the idea of the `GatewayStreamer#streamFor` having a side effect of broadcasting there are jobs available.

1. It really should be an engine side effect, especially once we start processing uncommitted records
2. It's not great that it's a side effect (functionally speaking) for a method which seems to be a getter, which not intuitive. I would propose we find some better way of making it explicit (but we can also possibly do this after)

**Comment by npepinpe** [comment]
CodeQL really doesn't like when you change bases :smile:

File: broker/src/main/java/io/camunda/zeebe/broker/transport/streamapi/JobStreamApiServer.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  public JobStreamApiServer(
+      final ClusterCommunicationService transport,
+      final StreamApiHandler<JobActivationProperties> requestHandler) {
+    this.transport = transport;
+    this.requestHandler = requestHandler;
+  }
+
+  @Override
+  public boolean isRelevant(final ClusterMembershipEvent event) {
```
Comment (deepthidevaki) [inline broker/src/main/java/io/camunda/zeebe/broker/transport/streamapi/JobStreamApiServer.java:36]: :+1: Didn't know this exists :D

**Review by deepthidevaki — APPROVED** [review]
Thanks. Just one small comment.

> Anyway, other notes: I'm not sure I like the idea of the GatewayStreamer#streamFor having a side effect of broadcasting there are jobs available.

I think we should re-visit it. It is not good to put long-polling and job push together. It is ok for this PR. May be we can revisit it when the engine part is done.

File: broker/src/main/java/io/camunda/zeebe/broker/transport/streamapi/JobStreamApiServer.java (hunk 1 above)
Comment (npepinpe) [reply broker/src/main/java/io/camunda/zeebe/broker/transport/streamapi/JobStreamApiServer.java:36]: Me neither, so I can't fully guarantee it works :smile:

**Comment by npepinpe** [comment]
bors merge

**Pull request merged**